[MSHARED-1128] Update documentation with new execute method (#49)
apache/maven-verifier: src/site/markdown/getting-started.md

## Run

> Calling `execute` runs Maven with the given Verifier configuration like, cli arguments, environment variables. 

Reviewer: ```suggestion
Calling `execute` runs Maven with the given Verifier configuration like CLI arguments and environment variables. 
```

Reviewer 2: The uppercase CLI should be everywhere in the document

Reviewer 2: configuration, like ...

Reviewer 2: CLI

> We must provide at lease one cli argument for executing.

Reviewer: ```suggestion
At least one CLI argument must be given which specifies either the Maven phase(s) and/or the goal(s) to execute.
```

Reviewer 2: Do we really need that? What if `/project/build/defaultGoal` is set?

Reviewer 2: must be provided

Author: Maybe:

``
At least one CLI argument should be provided which specifies either the Maven phase(s) and/or the goal(s) to execute unless project has default goal.
``

Reviewer 2: ...has a default goal.

Sounds good to me.

It throws a `VerificationException` in case the execution is not successful (e.g. binary not found or exit code > 0). 

> It is either using a forked JVM or is executed in the same JVM depending on the configuration.

Reviewer: ```suggestion
The method is using either a forked JVM or the same JVM for executing Maven depending on the configuration.
```

```
verifier.addCliArgument( "package" )
verifier.execute();
```

## Verify

> After executing the Maven process there are several methods starting with prefix `verify` 

Reviewer: ```suggestion
After calling `execute` one should call one or multiple of the methods starting with prefix `verify` to
```

> which allow you to check for the build result, check the log for certain contents and the existence of generated artifacts.

Reviewer: ```suggestion
* check for the build result
* check the log for certain contents or 
* check for the existence of generated artifacts.
```

> The main method `verify(boolean)` takes into consideration a file named `expected-results.txt` being located in the tested project base directory. 

Reviewer: ```suggestion
The main method `verify(boolean)` takes into consideration a file named `expected-results.txt` being located in the tested project's base directory. 
```

## Run

> Calling `execute` runs Maven with the given Verifier configuration, like CLI arguments and environment variables.

Reviewer 2: `execute()`
